tests/spec/urlparser-spec.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Test for URL Parameter parsing
+ */
+
+
+let defaultUrl = "https://python.microbit.org/v/1.1"; 
+describe("Testing different URL formats", () => {
+    test("Check if no Parameter is recognised", () =>{
+        expect(global.urlparse("parameter", defaultUrl) === undefined).toBeTruthy();
+    });
+    test("Check if URL fragments are ignored", ()=>{
+        expect(global.urlparse("anchor", defaultUrl + "#anchor") === undefined).toBeTruthy();
+    });
+    test("Check if parameter is recognised", () =>{
+        expect(global.urlparse("key", defaultUrl + "?key=value1") === "value1").toBeTruthy();
+    });
```

Comment:
Maybe we should also add tests for values in the URL set to `0`, `null`, `undefined`.

Reply:
should the value returned be a string "null" or actually null?